context/BasicContractContext.tsx

```diff
@@ -0,0 +1,65 @@
+import { useState, createContext, useContext, ReactNode, useEffect } from 'react'
+import { ethers, Contract } from 'ethers'
+import { Provider } from '@ethersproject/providers'
+import { WalletContext } from './WalletContext'
+
+// Uniswap contracts
+const { abi: IUniswapV2PairABI } = require('@uniswap/v2-core/build/IUniswapV2Pair.json')
+const { abi: IUniswapV2FactoryABI } = require('@uniswap/v2-core/build/IUniswapV2Factory.json')
+const { abi: IUniswapV2Router02ABI } = require('@uniswap/v2-periphery/build/IUniswapV2Router02.json')
+
+
+type BasicContractType = {
+    IUniswapV2PairContract: Contract | null
+    IUniswapV2FactoryContract: Contract | null
+    IUniswapV2Router02Contract: Contract | null
+}
+
+type BasicContractProviderProps = {
+    children: ReactNode
+}
+
+export const BasicContractContext = createContext<BasicContractType>({} as BasicContractType)
```

Comment:
This was originally to globalize the smart contract instantiation. So that we wouldn't have to do
```javascript
const someSmartContract = new ethers.Contract(address, contractABI, provider)
```
every time we want to use a smart contract.

However, I realized the ethers library makes it so that you have to instantiate it on the same file. Which is weird. Will do more research on this when we have Daniel's new contracts and/or open a new issue to solve this on Ethers's repo